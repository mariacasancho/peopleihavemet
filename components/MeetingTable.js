import React from 'react';
import { Table } from 'antd';

const MeetingTable = ({ meetings }) => {
    console.log(meetings);
    const columns = [
        {
            title: 'Person',
            dataIndex: 'person',
        },

        {
            title: 'Date',
            dataIndex: 'date',
        },
    ];

    const onChange = (pagination, filters, sorter, extra) => {
        console.log('params', pagination, filters, sorter, extra);
    };

    return (
        <Table columns={columns} dataSource={meetings} onChange={onChange} />
    );
};

export default MeetingTable;