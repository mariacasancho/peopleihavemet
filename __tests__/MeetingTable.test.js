import { render } from "@testing-library/react";
import MeetingTable from "../components/MeetingTable";


describe("MeetingTable Component", () => {
    Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(),
            removeListener: jest.fn(),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    it("should render properly", () => {
        const subject = render(<MeetingTable />);
        const tree = subject.container;
        expect(tree).toMatchSnapshot();
    });
});