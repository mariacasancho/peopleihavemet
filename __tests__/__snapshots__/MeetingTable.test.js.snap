// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingTable Component should render properly 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-empty"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-table-filter-column"
                      >
                        <span
                          class="ant-table-filter-column-title"
                        >
                          Person
                        </span>
                        <span
                          class="ant-table-filter-trigger-container"
                        >
                          <span
                            class="ant-dropdown-trigger ant-table-filter-trigger"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="search"
                              class="anticon anticon-search"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="search"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-table-filter-column"
                      >
                        <span
                          class="ant-table-filter-column-title"
                        >
                          Date
                        </span>
                        <span
                          class="ant-table-filter-trigger-container"
                        >
                          <span
                            class="ant-dropdown-trigger ant-table-filter-trigger"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="search"
                              class="anticon anticon-search"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="search"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-table-filter-column"
                      >
                        <span
                          class="ant-table-filter-column-title"
                        >
                          Location
                        </span>
                        <span
                          class="ant-table-filter-trigger-container"
                        >
                          <span
                            class="ant-dropdown-trigger ant-table-filter-trigger"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="search"
                              class="anticon anticon-search"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="search"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-placeholder"
                  >
                    <td
                      class="ant-table-cell"
                      colspan="3"
                    >
                      <div
                        class="ant-empty ant-empty-normal"
                      >
                        <div
                          class="ant-empty-image"
                        >
                          <svg
                            class="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                class="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                class="ant-empty-img-simple-g"
                                fill-rule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  class="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <p
                          class="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
